draft-thomson-ppm-l1-bound-sum.md

```diff
@@ -111,51 +111,126 @@ The difference is that Prio3L1BoundSum involves validation of
 which might increase the most efficient value for chunk_length.
 
 
-## Encoding and Decoding
+## Validity Circuit
 
-The encode, truncate, and decode functions for Prio3L1BoundSum is identical to that of Prio3SumVec;
-see {{Section 7.4.3 of VDAF}} for those definitions.
+The validity circuit for Prio3L1BoundSum uses a modified version of the
+Prio3SumVec validity circuit, see {{Section 7.4.3 of VDAF}}.
 
+The values from the measurement are extended to include their sum, so that the
+sum is checked in the same way as each vector component. The encoded measurement
+has a total length of `(length + 1) * bits`. The field elements in the encoded
+vector represent all the bits of the measurement vector's elements, followed by
+the bits of the L1 norm of the vector.
 
-## Validity Circuit
+The validity circuit checks that the encoded measurement consists of ones and
+zeros, and that the encoded L1 norm is consistent with the encoded vector
+elements. The first checks, that all field elements are ones and zeros, are done
+in the same manner as in the `SumVec` circuit. The L1 norm is checked by
+decoding the reported L1 norm, decoding the measurement vector, recomputing the
+L1 norm, and subtracting to confirm they are identical.
 
-The validity circuit for Prio3L1BoundSum uses a modified version of the Prio3SumVec validity circuit.
-The values from the measurement are extended to include their sum,
-so that the sum is checked in the same way as each vector component.
+The complete circuit is specified below.
 
 ~~~ python
-def eval(
-        self,
-        meas: list[F],
-        joint_rand: list[F],
-        num_shares: int) -> list[F]:
-    weight = 0
-    for i in range(self.length):
-        weight += self.field.decode_from_bit_vector(
-            meas[i * self.bits : (i + 1) * self.bits]
+class L1BoundSum(Valid[list[int], list[int], F]):
```

Comment:
So this is not going to render sensibly.  It seems like there is a preprocessing step used in the VDAF draft that this isn't able to use.

I can see how it would help to have this code somewhere, so that it could be run, but this draft is not going to be able to take the raw python.